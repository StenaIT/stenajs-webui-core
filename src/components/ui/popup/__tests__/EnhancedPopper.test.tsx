import { shallow } from 'enzyme';
import * as React from 'react';

// TODO: Fix current ts config settings and jest
it('works', () => {
  const Component = () => <div />;
  const wrapper = shallow(<Component />);
  expect(wrapper.length).toBe(1);
});

// import * as emotion from 'emotion';
// import { shallow } from 'enzyme';
// import { createSerializer } from 'jest-emotion';
// import * as React from 'react';
// import { ReactElement }Â from 'react';
// import * as Popper from 'react-popper';
// import { PopperChildrenProps } from 'react-popper';
// import { EnhancedPopper, PopperProps, PurePopper } from '../EnhancedPopper';
//
// expect.addSnapshotSerializer(createSerializer(emotion));
//
// describe('EnhancedPopper', () => {
//   const renderPropsProps: PopperChildrenProps = {
//     arrowProps: {
//       ref: () => null,
//       style: {},
//     },
//     outOfBoundaries: false,
//     placement: 'auto-end',
//     scheduleUpdate: () => null,
//     style: {},
//     ref: () => null,
//   };
//
//   const props: PopperProps = {
//     children: <div>children</div>,
//     PopperComponent: Popper.Popper,
//   };
//
//   it('renders PopperComponent', () => {
//     const wrapper = shallow(<EnhancedPopper {...props} />);
//     expect(wrapper.dive().type()).toBe(Popper.Popper);
//   });
//
//   it('styles targetContainer', () => {
//     const children = <div>children</div>;
//     const wrapper = shallow(<PurePopper {...props} children={children} />);
//     const renderPropsWrapper = shallow(wrapper.prop('children')(
//       renderPropsProps,
//     ) as ReactElement<{}>);
//     expect(
//       renderPropsWrapper
//         .childAt(0)
//         .dive()
//         .prop('children'),
//     ).toBe(children);
//   });
//
//   describe('styles', () => {
//     it('sets default styles', () => {
//       const wrapper = shallow(<PurePopper {...props} />);
//       const renderPropsWrapper = shallow(wrapper.prop('children')(
//         renderPropsProps,
//       ) as ReactElement<{}>);
//       expect(renderPropsWrapper.childAt(0).dive()).toMatchSnapshot();
//     });
//
//     it('uses passed styles', () => {
//       const backgroundColor = 'backgroundColor';
//       const shadow = false;
//       const targetMinHeight = 3;
//       const targetMinWidth = 2;
//       const wrapper = shallow(
//         <PurePopper
//           {...props}
//           backgroundColor={backgroundColor}
//           shadow={shadow}
//           targetMinHeight={targetMinHeight}
//           targetMinWidth={targetMinWidth}
//         />,
//       );
//       const renderPropsWrapper = shallow(wrapper.prop('children')(
//         renderPropsProps,
//       ) as ReactElement<{}>);
//       expect(renderPropsWrapper.childAt(0).dive()).toMatchSnapshot();
//     });
//   });
// });
