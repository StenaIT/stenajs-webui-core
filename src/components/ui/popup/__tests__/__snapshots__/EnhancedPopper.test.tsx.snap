// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnhancedPopper styles sets default styles 1`] = `
.emotion-0 {
  background-color: #fff;
  box-shadow: rgba(0,0,0,0.28) 0 2px 13px 0;
}

<div
  className="emotion-0"
>
  <div>
    children
  </div>
</div>
`;

exports[`EnhancedPopper styles uses passed styles 1`] = `
.emotion-0 {
  background-color: backgroundColor;
  min-height: 3px;
  min-width: 2px;
}

<div
  className="emotion-0"
>
  <div>
    children
  </div>
</div>
`;
